lib/FileCard/index.js

```diff
@@ -36,19 +96,9 @@ class FileCard extends Component {
         <div className="file-card__wrapper">
           <div className="file-card__label">{label}</div>
 
-          {previewSrc && (
-            <div
-              title={label}
-              style={{ backgroundImage: `url(${previewSrc})` }}
-              className="file-card__thumbnail"
-            />
-          )}
+          {previewSrc && this.backgroundImage()}
```

Comment:
You may want to change from `previewSrc` to `this.state.imageHasLoaded`. That way your render logic is all around when the image is ready or not.